handlers/main.yml

```diff
@@ -13,3 +13,6 @@
 - name: Restart mesos-slave
   service: name=mesos-slave state=restarted
   when: mesos_install_mode == "slave" or mesos_install_mode == "master-slave"
+
+- name: Clear mesos-slave cache
+  file: path=/var/lib/mesos/meta/slaves/latest state=absent
```

Comment:
This is probably the solution for https://github.com/AnsibleShipyard/ansible-mesos/issues/65 :)